indra/newview/llagent.cpp

```diff
@@ -4320,8 +4321,17 @@ void LLAgent::teleportViaLandmark(const LLUUID& landmark_asset_id)
 
 void LLAgent::doTeleportViaLandmark(const LLUUID& landmark_asset_id)
 {
-    LLViewerRegion *regionp = getRegion();
-    if(regionp && teleportCore())
+    bool is_local(false);
+    LLViewerRegion* regionp  = getRegion();
+
+    if (LLLandmark* landmark = gLandmarkList.getAsset(landmark_asset_id, NULL))
```

Comment:
I think this needs to check `region`, this is using `regionp->getHandle()` insde.